commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -236,13 +257,134 @@ object ReflectionUtil {
+            )
+        }
+      }
+      else if (ReflectionUtil.isSeqClass(resultType) || ReflectionUtil.isSetClass(resultType)) {
+        ReflectionUtil.newScalaCollection(resultType.asInstanceOf[Class[_ <: Iterable[_]]], typedValues)
+      }
+      else {
+        throw new ReflectionUtilException(s"Unknown collection type '${resultType.getSimpleName}'")
+      }
+    }
+    else if (value.size != 1) {
+      throw new ReflectionUtilException(s"Expecting a single argument to convert to ${resultType.getSimpleName} but got ${value.size}")
+    }
+    else {
+      buildUnitOrOptionFromString(resultType, unitType, value.head).get
+    }
+  }
+
+  /**
+    * Supports any class with a string constructor, [[Enum]], and [[java.nio.file.Path]].  If the class is an [[Option]],
+    * it will either return `None` if `s == null`, or `Some` object wrapping the an object of the wrapped type.
+    */
+  private def buildUnitOrOptionFromString(resultType: Class[_], unitType: Class[_], value: String): Try[Any] = Try {
+    val unit = buildUnitFromString(unitType, value).get
+    if (resultType == classOf[Option[_]]) Option(unit)
+    else unit
+  }
+
+  /** Attempts to construct a single value of unitType from the provided String and return it. */
+  private[reflect] def buildUnitFromString(unitType: Class[_], value: String) : Try[Any] = Try {
+    val clazz = ifPrimitiveThenWrapper(unitType)
+    if (clazz.isEnum) {
+      lazy val badArgumentString = s"'$value' is not a valid value for ${clazz.getSimpleName}. " +
+        enumOptions(clazz.asInstanceOf[Class[_ <: Enum[_ <: Enum[_]]]]).get
+      val maybeEnum = clazz.getEnumConstants.map(_.asInstanceOf[Enum[_]]).find(e => e.name() == value)
+      maybeEnum.getOrElse(throw new ReflectionUtilException(badArgumentString))
+    }
+    else if (clazz == classOf[Path]) {
+      StringUtil.pathTo(value)
+    }
+    else if (clazz == classOf[Option[_]]) {
+      // unitType shouldn't be an option, since it's supposed to the type *inside* any containers
+      throw new ReflectionUtilException(s"Cannot construct an '${unitType.getSimpleName}' from a string.  Is this an Option[Option[...]]?")
+    }
+    else if (clazz == classOf[java.lang.Object]) value
+    else {
+      import java.lang.reflect.Constructor
+      val ctor: Constructor[_] = clazz.getDeclaredConstructor(classOf[String])
+      ctor.setAccessible(true)
+      ctor.newInstance(value)
+    }
+  }
+
+  /**
+    * Returns the string "s" if x is greater than 1.
+    *
+    * @param x Value to test.
+    * @return "s" if x is greater than one else "".
+    */
+  private def plural(x: Int) = if (x > 1) "s" else ""
+
+  // Also used for Boolean Options
+  private[reflect] val EnumOptionDocPrefix: String = "Options: "
+  private[reflect] val EnumOptionDocSuffix: String = "."
+
+  /** Gets all the options for an enumeration.
+    */
+  def enumOptions(clazz: Class[_ <: Enum[_ <: Enum[_]]]): Try[String] = Try {
```

Comment:
How about having this just return a Seq[String] and moving the code to turn it into a single string for display be in the calling code?